### PR TITLE
[public-api] JWT Signer & Verifier WEB-101

### PR DESCRIPTION
**Pull request by easyCZ**
## Related Issue(s)

Fixes #

## How to test

1. public-api starts

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - mp-papi-jw18dfbeb071</li>
	<li><b>🔗 URL</b> - <a href="https://mp-papi-jw18dfbeb071.preview.gitpod-dev.com/workspaces" target="_blank">mp-papi-jw18dfbeb071.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
</ul>

## Build Options:

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`

<details>
<summary>Publish Options</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer Options</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] with-ws-manager-mk2
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

#### Preview Environment Options:
- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

/hold

### CODE REVIEW

File: components/public-api-server/pkg/server/server.go [hunk 1]
```diff
@@ -126,6 +126,11 @@ func Start(logger *logrus.Entry, version string, cfg *config.Configuration) erro
 		log.Info("No Personal Access Token signign key specified, PersonalAccessToken service will be disabled.")
 	}
 
+	_, err = auth.NewJWTFromAuthPKI(cfg.Auth.PKI, 7*24*time.Hour, "TODO")
```
Comment (easyCZ) [inline components/public-api-server/pkg/server/server.go:129]: For now, we don't need to set the issuer as we're not actually issuing any JWTs. 

What we'll want to do here is match the issuer on "server" - gitpod domain.

Comment (easyCZ) [reply components/public-api-server/pkg/server/server.go:129]: To supply consistent values, there's https://github.com/gitpod-io/gitpod/pull/17314 which will use the same config between server/public-api

**Review by geropl — APPROVED** [review]
:tada: Changes LGTM, and papi is running :heavy_check_mark:

**Pull request merged**